src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -615,13 +615,18 @@ public void Dispose()
             SpanProcessor spanProcessor,
             Resource libraryResource)
         {
+            var spanCreationOptions = new SpanCreationOptions
+            {
+                Attributes = activity.Tags?.ToDictionary(k => k.Key, v => (object)v.Value),
```

Comment:
activity.Tags always return non-null value. It would be better to test the count instead. 

```C$
var tags = activity.Tags;
if (tags.Count() > 0)
{
....
}
```

Reply:
@SergeyKanzhelev why we have SpanCreationOptions.Attributes defined as IDictionary<string, object> instead of `IEnumerable<KeyValuePait<string, object>>`? I am asking because we can avoid allocating the intermediate IDictionary object and pass IEnumerable directly which we got from Activity.Tags.

Reply:
Thanks for the quick review, will make the changes later then.

About the dictionary I also had that question, wondering if it is to guarantee the uniqueness of the keys?

Reply:
> wondering if it is to guarantee the uniqueness of the keys?

Yes, the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes) is saying the keys should be unique. but at the same time, it requires keeping the order of the keys too which the Dictionary doesn't guarantee that.

```
Attributes SHOULD preserve the order in which they're set. Setting an attribute with the same key as an existing attribute SHOULD overwrite the existing attribute's value.
```

Reply:
Yes thats true, and after checking some parts of the code dont see any specific reason to be a dictionary. Like you stated the keys are being overwriten when they are repeated and the `IEnumerable` would avoid unecessary memory allocation in cases like this PR.

Also I made the changes that you mentioned